File: backend/DTO/FacultyDTO.cs
using System.Text.Json.Serialization;

namespace backend.DTO;

public record FacultyDTO
{
    [JsonPropertyName("id")]
    public string? ID { get; init; }
    [JsonPropertyName("code")]
    public required string Code { get; init; }
    [JsonPropertyName("name")]
    public required string Name { get; init; }
    [JsonPropertyName("count")]
    public int Count { get; init; } = 0;

    [JsonPropertyName("created_at")]
    public DateTime? CreatedAt { get; init; }
}

File: frontend/src/api/dto/faculty.dto.ts
import { Faculty } from "@/src/types";

export type FacultyDTO = {
  id: string;
  code: string;
  name: string;
  count: number;
  created_at: string;
};

export const parseFacultyDTO = (data: FacultyDTO): Faculty => ({
  id: data.id,
  code: data.code,
  name: data.name,
  count: data.count,
  createdAt: data.created_at,
});

export const parseFacultyDTOList = (data: FacultyDTO[]): Faculty[] => {
  return data.map(parseFacultyDTO);
};
